### PR TITLE
Create the third tab "Project participants" of the "Create Project" view

### PR DESCRIPTION
**Pull request by denektenina**
# Description

Link to Jira issue: [NP-47305](https://sikt.atlassian.net/browse/NP-47305)

**⚠This is only visible when you have beta toggled to true in local storage ⚠**

You get to this view my going to Min Side --> "Prosjektregistreringer" in the left panel -> Choose a project -> Click the edit-button and go to the third tab: Contributors

This is a part of the effort to create a new create project-wizard [(NP-47281)](https://sikt.atlassian.net/browse/NP-47281), and there is a subtask for each of the 4 steps, plus subtasks for some new functionality to add.

In this task we have to create the third tab, using already the new contributor view I have made in earlier tasks.

**What should work after this PR:**

In projects view, it should now be possible to click the edit button and get into the new edit project wizard and navigate to the third panel:

image

* It should be possible to see the current contributors and their affiliations, and change them.
* You can click on the stepper to change views (the forth view is empty)
* This new view should only be visible when beta is toggled to true in localStorage, so verify that it looks as before when this is not the case.
* Error check between tabs

**Things that are not working and will be fixed in later PRs:**
* The last panel is missing
* There is no save-button
* There is no cancel-button
* There is no-next- or previous button
* The stepper is not ready for mobile-view yet
* There is still nothing when you open the new-project view

Note: There will be a new design for project contributors where they are divided into roles instead, so this will all be changed pretty soon. But since this design is already approved and ready from recent iterations in the non-beta-version, it can be used for now.

# Checklist

Ensure that the changes are aligned with the expectations of PO/designer before marking the PR as ready for review.

Also ensure that the following criterias are met:

- [x] The changes are working as expected
- [x] The changes are tested OK for different screen sizes
- [x] The changes are tested OK for a11y
- [x] Interactive elements have data-testids
- [x] I have done a QA of my own changes

_Note: some of these criterias may not always be relevant, and can simply be marked as completed._

[NP-47305]: https://sikt.atlassian.net/browse/NP-47305?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/pages/project/project_wizard/ProjectForm.tsx [hunk 1]
```diff
@@ -22,6 +22,7 @@ interface ProjectFormProps {
 export const ProjectForm = ({ project }: ProjectFormProps) => {
   const { t } = useTranslation();
   const [tabNumber, setTabNumber] = useState(ProjectTabs.Description);
+  const [maxVisitedTab, setMaxVisitedTab] = useState(ProjectTabs.Description);
```
Comment (elkebab) [inline src/pages/project/project_wizard/ProjectForm.tsx:25]: Feel free til å vurdere om det hadde vært enklere å plassere sånne ting i react-context for å slippe å sende som prop mange plasser også!
Var enda enklere enn jeg håpte på når jeg gjorde det for de NVI-feltene iaf: https://github.com/BIBSYSDEV/NVA-Frontend/pull/6093
Men det er absolutt ikke noe must nå

File: src/pages/projects/form/ProjectContributors.tsx
Comment (elkebab) [inline src/pages/projects/form/ProjectContributors.tsx:1]: Noen småting som hikker når man gjør ting ser jeg, men lever veldig fint med det nå, og at heller tar en skikkelig runde på det til slutt

https://github.com/user-attachments/assets/5783b6fd-76ef-49d4-a119-c3c5728c1f67

**Review by elkebab — APPROVED** [review]
Har testet bittelitt, men lever som sagt i annen kommentar fint med å ha en skikkelig runde testing av dette når alt er flyttet over. Uansett nok et langt steg frem dette!

File: src/utils/formik-helpers/project-form-helpers.ts [hunk 2]
```diff
@@ -16,6 +21,7 @@ export const getProjectTabErrors = (errors: FormikErrors<CristinProject>, touche
       touched
     ),
     [ProjectTabs.Details]: getErrorMessages<CristinProject>(Object.values(ProjectDetailsFieldNames), errors, touched),
+    [ProjectTabs.Contributors]: getErrorMessages<CristinProject>(Object.values(ProjectContributorsFieldNames), errors),
```
[the remark this answers is not in the document]
Comment (denektenina) [reply src/utils/formik-helpers/project-form-helpers.ts:24]: Yes! Jeg bruker ikke formik sin touched-logikk her, og en av dere som kodet dette er har forutseende nok til å gjøre det valgfritt!

File: src/pages/projects/form/ProjectContributors.tsx [hunk 3]
```diff
@@ -103,7 +107,7 @@ export const ProjectContributors = ({ suggestedProjectManager }: ProjectContribu
                 </TableContainer>
               </ListPagination>
             )}
-            {contributorError && typeof contributorError === 'string' && touched.contributors && (
+            {contributorError && typeof contributorError === 'string' && isVisited && (
```
[the remark this answers is not in the document]
Comment (denektenina) [reply src/pages/projects/form/ProjectContributors.tsx:110]: Yes, det er for å gjøre den mer generisk så jeg kunne bruke den i en ny usecase. Jeg tror kanskje jeg fant en måte å bruke tab'sene for å sjekke om feilmelding skal vises istedet for å måtte manuelt sette ting til touched. Ble ihvertfall mindre kode enn i den gamle versjonen :D

File: src/pages/projects/form/ProjectContributors.tsx
Comment (denektenina) [reply src/pages/projects/form/ProjectContributors.tsx:1]: Skriver det ned og tar det fortløpende!

**Pull request merged**